modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/semanticui/elements/popup/Popup.scala

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package edu.gemini.seqexec.web.client.semanticui.elements.popup
+
+import japgolly.scalajs.react.component.Scala.Unmounted
+import japgolly.scalajs.react.{Callback, ScalaComponent}
+import japgolly.scalajs.react.vdom.html_<^._
+import japgolly.scalajs.react.raw.ReactElement
+
+/**
+  * Produces a popup using javascript
+  * This is a barebones component waiting for the proper react component to be made available
+  */
+@SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements", "org.wartremover.warts.AsInstanceOf"))
+object Popup {
+  @SuppressWarnings(Array("org.wartremover.warts.DefaultArguments"))
+  final case class Props(selector: String, content: String)
+
+  private val component = ScalaComponent.builder[Props]("Popup")
+    .stateless
+    .renderPC{($, p, c) =>
+      // This is in principle unsafe but we are only allowing Elements on the constructor
+      VdomElement($.propsChildren.only().asInstanceOf[ReactElement])
+    }
+    .componentDidMount(ctx =>
+      Callback {
+        // Enable menu on Semantic UI
+        import org.querki.jquery.$
+        import edu.gemini.web.client.facades.semanticui.SemanticUIPopup._
+
+        $(ctx.getDOMNode).popup(
+          JsPopupOptions.content(ctx.props.content)
+        )
+      }
+    ).build
+
+  def apply(p: Props, children: VdomElement): Unmounted[Props, Unit, Unit] = component(p)(children)
+}
```

Comment:
Could you explain a little bit what is happening here? Do we know why the original CSS solution wasn't working?

So now, for each button (or UI element) with a popup, we create a stateless `ScalaComponent`.
When is the `Callback` called?  And if you could explain lines 32-33, that would be helpful :-).

Reply:
Yes, the original pure CSS solution wasn't able to display beyond the target leading to clipped popups.

in line 38. `children` is a reference to what we are wrapping, typically a button, (note that i'm limiting this to only one child)

`SemanticUI` uses jQuery. the code in line 32 creates a `jQuery` reference to the rendered node (In this case the child) and attaches a popup with the `popup` call (a `jQuery` extension) and it also passes some configuration. In particular the text associated with the popup

in general all the code using `SemanticUI` and `jQuery` is fairly low level and hacky. This will improve once i complete the facades for `Semantic UI React`

Reply:
`Callback` is called by `scalajs-react`. It is slightly similar to `IO` in `cats` or `scalaz`. It lets us return pure functions from the lifecycle methods and it is pervasive on `scalajs-react`

Reply:
That helps a lot. Thanks!